src/test/java/uk/gov/companieshouse/transactions/web/controller/confirmation/ConfirmationControllerTests.java

```diff
@@ -0,0 +1,126 @@
+                                                        "/confirmation";
+
+    private static final String CONFIRMATION_VIEW = "transactionConfirmation";
+
+    private static final String CONFIRMATION_MODEL_ATTR = "confirmation";
+
+    private static final String ERROR_VIEW = "error";
+
+    @BeforeEach
+    private void setup() {
+
+        this.mockMvc = MockMvcBuilders.standaloneSetup(controller).build();
+    }
+
+    @Test
+    @DisplayName("Get confirmation view - success path")
+    void getConfirmationSuccess() throws Exception {
+
+        Transaction closedTransaction = new Transaction();
+
+        when(transactionsService.getTransaction(TRANSACTION_ID)).thenReturn(closedTransaction);
+
+        when(transactionsService.isTransactionClosed(closedTransaction)).thenReturn(true);
+
+        when(confirmationService.getTransactionConfirmation(closedTransaction))
+                .thenReturn(new Confirmation());
+
+        this.mockMvc.perform(get(CONFIRMATION_PATH))
+                .andExpect(view().name(CONFIRMATION_VIEW))
+                .andExpect(status().isOk())
+                .andExpect(model().attributeExists(CONFIRMATION_MODEL_ATTR));
+    }
+
+    @Test
+    @DisplayName("Get confirmation view - transaction open")
+    void getConfirmationForOpenTransaction() throws Exception {
+
+        Transaction openTransaction = new Transaction();
+
+        when(transactionsService.getTransaction(TRANSACTION_ID)).thenReturn(openTransaction);
+
+        when(transactionsService.isTransactionClosed(openTransaction)).thenReturn(false);
+
+        this.mockMvc.perform(get(CONFIRMATION_PATH))
+                .andExpect(view().name(ERROR_VIEW))
+                .andExpect(status().isOk());
+
+        verify(confirmationService, never()).getTransactionConfirmation(any());
+    }
+
+    @Test
+    @DisplayName("Get confirmation view - ServiceException thrown on getTransaction")
+    void getConfirmationServiceExceptionThrowOnGetTransaction() throws Exception {
+
+        when(transactionsService.getTransaction(TRANSACTION_ID)).thenThrow(ServiceException.class);
+
+        this.mockMvc.perform(get(CONFIRMATION_PATH))
+                .andExpect(view().name(ERROR_VIEW))
+                .andExpect(status().isOk());
+
+        verify(confirmationService, never()).getTransactionConfirmation(any());
+    }
+
+    @Test
+    @DisplayName("Get confirmation view - ServiceException thrown on getTransactionConfirmation")
+    void getConfirmationServiceExceptionThrowOnGetTransactionConfirmation() throws Exception {
+
+        Transaction closedTransaction = new Transaction();
+
+        when(transactionsService.getTransaction(TRANSACTION_ID)).thenReturn(closedTransaction);
+
+        when(transactionsService.isTransactionClosed(closedTransaction)).thenReturn(true);
+
+        when(confirmationService.getTransactionConfirmation(closedTransaction)).thenThrow(ServiceException.class);
```

Comment:
can we do an `assertThrows` on the `getTransactionConfirmation()` method just to make sure it definitely did throw a `ServiceException`?

Reply:
Done